src/platforms/Points/EditPointsModal.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+} from "@chakra-ui/react"
+import useEditGuildPlatform from "components/[guild]/AccessHub/hooks/useEditGuildPlatform"
+import { AddPointsFormType } from "components/[guild]/RolePlatforms/components/AddRoleRewardModal/components/AddPointsPanel/AddPointsPanel"
+import AddNewPointsType from "components/[guild]/RolePlatforms/components/AddRoleRewardModal/components/AddPointsPanel/components/AddNewPointsType"
+import Button from "components/common/Button"
+import { Modal } from "components/common/Modal"
+import useToast from "hooks/useToast"
+import { useEffect } from "react"
+import { FormProvider, useForm, useWatch } from "react-hook-form"
+import { GuildPlatform } from "types"
+
+type Props = {
+  isOpen: boolean
+  onClose: () => void
+  guildPlatformId: number
+  platformGuildData: GuildPlatform["platformGuildData"]
+}
+
+const EditPointsModal = ({
+  isOpen,
+  onClose,
+  guildPlatformId,
+  platformGuildData,
+}: Props) => {
+  const { name: currentName, imageUrl: currentImage } = platformGuildData
+
+  const toast = useToast()
+  const { onSubmit, isLoading } = useEditGuildPlatform({
+    guildPlatformId,
+    onSuccess: () => {
+      toast({
+        status: "success",
+        title: "Successfully updated reward",
+      })
+      onClose()
+    },
+  })
+
+  const onEditPointsRewardSubmit = (data) => {
+    onSubmit({ platformGuildData: data })
+  }
+
+  const methods = useForm<AddPointsFormType>({
+    mode: "all",
+    defaultValues: {
+      name: currentName,
+      imageUrl: currentImage,
+    },
+  })
+
+  const { control } = methods
+  const localName = useWatch({ control, name: "name" })
+  const localImage = useWatch({ control, name: "imageUrl" })
+
+  useEffect(() => {
+    if (isOpen) methods.reset({ name: currentName, imageUrl: currentImage })
```

Comment:
Consider passing this to the `onClose` prop of the modal, that way we don't need to check if `isOpen` is true, instead we'll just reset the form when the user closes the modal:

```tsx
<Modal isOpen={isOpen} onClose={() => {
  onClose()
  methods.reset(...)
}}
...
>
```